x-pack/plugins/fleet/common/services/validate_package_policy.ts

```diff
@@ -46,6 +48,12 @@ export type PackagePolicyValidationResults = {
   inputs: Record<PackagePolicyInput['type'], PackagePolicyInputValidationResults> | null;
 } & PackagePolicyConfigValidationResults;
 
+export const DATASET_REGEX = /^(?![_\-\.])[a-z0-9_\-\.]+$/;
```

Comment:
Created a regex with the most common characters, the full set of rules allowed for index names are more complex:
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params

Reply:
Removed `-` from the allowed characters as it can cause another issue: https://github.com/elastic/kibana/issues/182952
This is consistent with the existing UI text saying `Set the name for your dataset. Changing the dataset will send the data to a different index. You can't use - in the name of a dataset and only valid characters for Elasticsearch index names.`